test-lib.sh

```diff
@@ -0,0 +1,142 @@
+#
+# Test a container image.
+#
+# Always use sourced from a specific container testfile 
+#
+# reguires definition of CID_FILE_DIR
+# CID_FILE_DIR=$(mktemp --suffix=<container>_test_cidfiles -d)
+# reguires definition of TEST_LIST 
+# TEST_LIST="\
+# run_container_creation_tests
+# run_doc_test <words_to_look_for_in_the_doc>"
+
+# Container CI tests
+# abbreviated as "concit"
+
+# may be redefined in the specific container testfile
+EXPECTED_EXIT_CODE=0
+
+function concit_cleanup() {
+  for cid_file in $CID_FILE_DIR/* ; do
+    CONTAINER=$(cat $cid_file)
+
+    : "Stopping and removing container $CONTAINER..."
+    docker stop $CONTAINER
+    exit_status=$(docker inspect -f '{{.State.ExitCode}}' $CONTAINER)
+    if [ "$exit_status" != "$EXPECTED_EXIT_CODE" ]; then
+      : "Dumping logs for $CONTAINER"
+      docker logs $CONTAINER
+    fi
+    docker rm $CONTAINER
+    rm $cid_file
+  done
+  rmdir $CID_FILE_DIR
+  : "Done."
+}
+
+function concit_enable_cleanup() {
+  trap concit_cleanup EXIT SIGINT
+}
+
+function concit_get_cid() {
+  local name="$1" ; shift || return 1
+  echo $(cat "$CID_FILE_DIR/$name")
+}
+
+function concit_get_cip() {
+  local id="$1" ; shift
+  docker inspect --format='{{.NetworkSettings.IPAddress}}' $(concit_get_cid "$id")
+}
+
+function concit_wait_for_cid() {
+  local max_attempts=10
+  local sleep_time=1
+  local attempt=1
+  local result=1
+  while [ $attempt -le $max_attempts ]; do
+    [ -f $cid_file ] && [ -s $cid_file ] && break
+    : "Waiting for container start..."
+    attempt=$(( $attempt + 1 ))
+    sleep $sleep_time
+  done
+}
+
+# Make sure the invocation of docker run fails.
+function concit_assert_container_creation_fails() {
+
+  # Time the docker run command. It should fail. If it doesn't fail,
+  # container will keep running so we kill it with SIGKILL to make sure
+  # timeout returns a non-zero value.
+  set +e
+  timeout -s SIGTERM --preserve-status 10s docker run --rm "$@" $IMAGE_NAME
+  ret=$?
+  set -e
+
+  # Timeout will exit with a high number.
+  if [ $ret -gt 30 ]; then
+    return 1
+  fi
+}
+
+# to pass some arguments you need to specify CONTAINER_ARGS variable
+function concit_create_container() {
+  cid_file="$CID_FILE_DIR/$1" ; shift
+  # create container with a cidfile in a directory for cleanup
+  docker run ${CONTAINER_ARGS:-} --cidfile="$cid_file" -d "$@" $IMAGE_NAME
+  : "Created container $(cat $cid_file)"
+  concit_wait_for_cid
+}
+
+function concit_scl_usage() {
```

Comment:
If you wanted a function `grep_container_command_output` you'd probably run the docker only once and not 3 times so therefore keeping the name. Once there is a better implementation of testing scl_usage it could be replaced.